### PR TITLE
Data Platform OMAS - initial commit

### PR DESCRIPTION
**Pull request by cong78**
Hi, 

This is the initial commit of Data Platform OMAS that is working on ingesting view creation events from Data Virtualization technologies. 
The codes are based on IV OMAS as previous view creation job will be done by it instead of Data Platform OMAS.

Please leave your feedback and comment. :)

Thanks,

Cong

### CODE REVIEW

File: open-metadata-implementation/access-services/data-platform/data-platform-api/src/main/java/org/odpi/openmetadata/accessservices/dataplatform/events/DataViewColumn.java [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonInclude;
+
+import static com.fasterxml.jackson.annotation.JsonAutoDetect.Visibility.NONE;
+import static com.fasterxml.jackson.annotation.JsonAutoDetect.Visibility.PUBLIC_ONLY;
+
+@JsonAutoDetect(getterVisibility = PUBLIC_ONLY, setterVisibility = PUBLIC_ONLY, fieldVisibility = NONE)
+@JsonInclude(JsonInclude.Include.NON_NULL)
+@JsonIgnoreProperties(ignoreUnknown = true)
+public class DataViewColumn extends DataViewElement {
```
Comment (ruxandraRosu) [inline open-metadata-implementation/access-services/data-platform/data-platform-api/src/main/java/org/odpi/openmetadata/accessservices/dataplatform/events/DataViewColumn.java]: Is this class used by the flow that creates the view? It should be used by the external tools as part of saving a view

File: open-metadata-implementation/access-services/data-platform/data-platform-api/src/main/java/org/odpi/openmetadata/accessservices/dataplatform/events/DataViewColumnSource.java [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+@JsonAutoDetect(getterVisibility = PUBLIC_ONLY, setterVisibility = PUBLIC_ONLY, fieldVisibility = NONE)
+@JsonInclude(JsonInclude.Include.NON_NULL)
+@JsonIgnoreProperties(ignoreUnknown = true)
+@JsonTypeInfo(
+        use = JsonTypeInfo.Id.NAME,
+        include = JsonTypeInfo.As.PROPERTY,
+        property = "class")
+public class DataViewColumnSource extends Source {
+
```
Comment (ruxandraRosu) [inline open-metadata-implementation/access-services/data-platform/data-platform-api/src/main/java/org/odpi/openmetadata/accessservices/dataplatform/events/DataViewColumnSource.java]: Similar to previous comment, is this used?

File: open-metadata-implementation/access-services/data-platform/data-platform-api/src/main/java/org/odpi/openmetadata/accessservices/dataplatform/events/DataViewElement.java [hunk 3]
```diff
@@ -0,0 +1,82 @@
+/* SPDX-License-Identifier: Apache-2.0 */
```
Comment (ruxandraRosu) [inline open-metadata-implementation/access-services/data-platform/data-platform-api/src/main/java/org/odpi/openmetadata/accessservices/dataplatform/events/DataViewElement.java]: Similar to previous comment, is this used?

File: open-metadata-implementation/access-services/data-platform/data-platform-api/src/main/java/org/odpi/openmetadata/accessservices/dataplatform/events/DataViewSource.java [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+
+@JsonAutoDetect(getterVisibility = PUBLIC_ONLY, setterVisibility = PUBLIC_ONLY, fieldVisibility = NONE)
+@JsonInclude(JsonInclude.Include.NON_NULL)
+@JsonIgnoreProperties(ignoreUnknown = true)
+@JsonTypeInfo(
+        use = JsonTypeInfo.Id.NAME,
+        include = JsonTypeInfo.As.PROPERTY,
+        property = "class")
+public class DataViewSource extends Source {
+
```
Comment (ruxandraRosu) [inline open-metadata-implementation/access-services/data-platform/data-platform-api/src/main/java/org/odpi/openmetadata/accessservices/dataplatform/events/DataViewSource.java]: Similar to previous comment, is this used?

File: open-metadata-implementation/access-services/data-platform/data-platform-api/src/main/java/org/odpi/openmetadata/accessservices/dataplatform/events/DataViewTable.java [hunk 5]
```diff
@@ -0,0 +1,38 @@
+/* SPDX-License-Identifier: Apache-2.0 */
```
Comment (ruxandraRosu) [inline open-metadata-implementation/access-services/data-platform/data-platform-api/src/main/java/org/odpi/openmetadata/accessservices/dataplatform/events/DataViewTable.java]: Similar to previous comment

File: open-metadata-implementation/access-services/data-platform/data-platform-server/src/main/java/org/odpi/openmetadata/accessservices/dataplatform/utils/Constants.java [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+    public static final String CONNECTION = "Connection";
+    public static final String SOFTWARE_SERVER = "SoftwareServer";
+    public static final String DATABASE_SERVER = "DatabaseServer";
+    public static final String SERVER_ENDPOINT = "ServerEndpoint";
+    public static final String SOFTWARE_SERVER_CAPABILITY = "SoftwareServerCapability";
+    public static final String ENGINE = "Engine";
+    public static final String REPORTING_ENGINE = "ReportingEngine";
+    public static final String PRIMARY_KEY = "PrimaryKey";
+    public static final String FOREIGN_KEY = "ForeignKey";
+    public static final String DEPLOYED_REPORT = "DeployedReport";
```
Comment (ruxandraRosu) [inline open-metadata-implementation/access-services/data-platform/data-platform-server/src/main/java/org/odpi/openmetadata/accessservices/dataplatform/utils/Constants.java]: I think some of the constants are not used and could be removed

File: open-metadata-implementation/access-services/data-platform/data-platform-server/src/main/java/org/odpi/openmetadata/accessservices/dataplatform/utils/QualifiedNameUtils.java [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+
+
+    public static String buildQualifiedNameForRelationalDbSchemaType(String hostAddress, String databaseName, String schemaName){
+        String endpointQualifiedName = QualifiedNameUtils.buildQualifiedName("", Constants.SOFTWARE_SERVER, hostAddress);
+        String databaseQualifiedName = QualifiedNameUtils.buildQualifiedName(endpointQualifiedName, Constants.DATA_STORE, databaseName);
+        return QualifiedNameUtils.buildQualifiedName(databaseQualifiedName, Constants.RELATIONAL_DB_SCHEMA_TYPE, schemaName + Constants.TYPE_SUFFIX);
+    }
+
+
+    public static String buildQualifiedNameForDataViewColumn(String hostAddress, String informationViewId, String columnName){
```
Comment (ruxandraRosu) [inline open-metadata-implementation/access-services/data-platform/data-platform-server/src/main/java/org/odpi/openmetadata/accessservices/dataplatform/utils/QualifiedNameUtils.java]: I think this method is not used in the flow for creating the view

File: open-metadata-implementation/access-services/data-platform/data-platform-api/src/main/java/org/odpi/openmetadata/accessservices/dataplatform/events/DataViewColumn.java (hunk 1 above)
Comment (cong78) [reply open-metadata-implementation/access-services/data-platform/data-platform-api/src/main/java/org/odpi/openmetadata/accessservices/dataplatform/events/DataViewColumn.java]: True. Removed.

File: open-metadata-implementation/access-services/data-platform/data-platform-api/src/main/java/org/odpi/openmetadata/accessservices/dataplatform/events/DataViewColumnSource.java (hunk 2 above)
Comment (cong78) [reply open-metadata-implementation/access-services/data-platform/data-platform-api/src/main/java/org/odpi/openmetadata/accessservices/dataplatform/events/DataViewColumnSource.java]: True. Removed.

File: open-metadata-implementation/access-services/data-platform/data-platform-api/src/main/java/org/odpi/openmetadata/accessservices/dataplatform/events/DataViewElement.java (hunk 3 above)
Comment (cong78) [reply open-metadata-implementation/access-services/data-platform/data-platform-api/src/main/java/org/odpi/openmetadata/accessservices/dataplatform/events/DataViewElement.java]: True. Removed.

File: open-metadata-implementation/access-services/data-platform/data-platform-api/src/main/java/org/odpi/openmetadata/accessservices/dataplatform/events/DataViewSource.java (hunk 4 above)
Comment (cong78) [reply open-metadata-implementation/access-services/data-platform/data-platform-api/src/main/java/org/odpi/openmetadata/accessservices/dataplatform/events/DataViewSource.java]: True. Removed.

File: open-metadata-implementation/access-services/data-platform/data-platform-api/src/main/java/org/odpi/openmetadata/accessservices/dataplatform/events/DataViewTable.java (hunk 5 above)
Comment (cong78) [reply open-metadata-implementation/access-services/data-platform/data-platform-api/src/main/java/org/odpi/openmetadata/accessservices/dataplatform/events/DataViewTable.java]: True. Removed.

File: open-metadata-implementation/access-services/data-platform/data-platform-server/src/main/java/org/odpi/openmetadata/accessservices/dataplatform/utils/Constants.java (hunk 6 above)
Comment (cong78) [reply open-metadata-implementation/access-services/data-platform/data-platform-server/src/main/java/org/odpi/openmetadata/accessservices/dataplatform/utils/Constants.java]: True. Removed. And also some other constants.

File: open-metadata-implementation/access-services/data-platform/data-platform-server/src/main/java/org/odpi/openmetadata/accessservices/dataplatform/utils/QualifiedNameUtils.java (hunk 7 above)
Comment (cong78) [reply open-metadata-implementation/access-services/data-platform/data-platform-server/src/main/java/org/odpi/openmetadata/accessservices/dataplatform/utils/QualifiedNameUtils.java]: True. Removed.

**Pull request merged**